plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeleteResourceHandler.java

```diff
@@ -48,16 +49,24 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 			return OpenShiftUIActivator.statusFactory().cancelStatus("No resource selected that we can delete."); //$NON-NLS-1$
 		}
 		try (Stream<IResourceUIModel> stream = Arrays.stream(resources)) {
-	        boolean hasProject = stream.anyMatch(resource -> resource instanceof IProjectAdapter);
+	        boolean willDeleteSubResources = stream.anyMatch(resource -> Platform.getAdapterManager().getAdapter(resource.getResource(), ResourceProcessor.class).willCascadeDeleteLinkedResources());
```

Comment:
Service has no linked resource as per the oc client. See my spec in JIRA: https://issues.jboss.org/browse/JBIDE-22241?focusedCommentId=13247145&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13247145